CONTRIBUTING.md

```diff
@@ -1,3 +1,169 @@
+| `refactor` | The `refactor` type is used to identify development changes related to modifying the codebase, which neither adds a feature nor fixes a bug - such as removing redundant code, simplifying the code, renaming variables, etc. |
+| `revert`   | For commits that revert one or more previous commits.                                                                                                                                                                         |
+| `style`    | The `style`type is used to identify development changes related to styling the codebase, regardless of the meaning  such as indentations, semi-colons, quotes, trailing commas and so on.                                     |
+| `test`     | The `test` type is used to identify development changes related to tests - such as refactoring existing tests or adding new tests.                                                                                            |
+
+In order to ensure that the format of your commit message adheres to the
+_Conventional Commits_ specification and the defined type vocabulary, you can
+use the [dedicated linter][conv-commits-lint]. More information about
+_Conventional Commits_ can also be found in this [blog
+post][conv-commits-blog].
+
+## Merging Your Code
+
+Here is a check list that you can follow to make sure that code merges happen
+smoothly:
+
+1. [Open an issue](#submitting-issues) _first_ to give other contributors a
+   chance to discuss the proposed changes (alternatively: assign yourself to
+   one of the existing issues)
+2. Clone the repository, create a feature branch off of the default branch
+   (never commit changes to protected branches directly) and implement your
+   code changes
+3. If applicable, update relevant sections of the [main documentation][docs]
+   and/or the [pipeline documentation][docs-pipeline]
+4. Add or update tests; untested code will not be merged; refer to the
+   [guidelines](#code-style-and-testing) above for details
+5. Ensure that your coding style is in line with the
+   [guidelines](#code-style-and-testing) described above
+6. Ensure that all tests and linter checks configured in the
+   [continuous integration][ci-cd] (CI) pipeline passes without issues
+7. If necessary, clean up excessive commits with `git rebase`; cherry-pick and
+   merge commits as you see fit; use concise and descriptive commit messages
+8. Push your clean, tested and documented feature branch to the remote; make
+   sure the [CI][ci-cd] pipeline passes
+9. Issue a pull request against the default branch; follow the instructions in
+   the [template][pull-request]; importantly, describe your changes in detail,
+   yet with concise language, and do not forget to indicate which issue(s)
+   the code changes resolve or refer to; assign a project maintainer to review
+   your changes
+
+
+## Becoming a Co-maintainer
```

Comment:
Should be: Becoming a Co-Maintainer, with a capital M. It's the more standard way of capitalizing compound words in title case.